lib/PuppeteerSharp/JSHandle.cs

```diff
@@ -61,7 +61,7 @@ public JSHandle(ExecutionContext context, Session client, object remoteObject)
             return (T)Helper.ValueFromRemoteObject<T>(RemoteObject);
         }
 
-        public virtual ElementHandle AsElement() => null;
+        public virtual ElementHandle AsElement() => this as ElementHandle;
```

Comment:
pretty sure this isn't needed since `evaluateHandle` creates a handle based on the response from the server, so if the response is an element it will return an element - https://github.com/GoogleChrome/puppeteer/blob/v1.0.0/lib/FrameManager.js#L199

Reply:
But we would be forcing the user to cast to `ElementHandle` in situations like this https://github.com/kblok/puppeteer-sharp/pull/205/files#diff-314af19ddd7e6f4f83af0ebf25e8ba80R24. 

How be that better? I think this won't hurt us.

Reply:
what if we change the signature of the method to something like this
```cs
public async Task<THandle> EvaluateExpressionHandleAsync<THandle>(string script) where THandle : JSHandle
```
even now the user calls `.AsElement`

Reply:
I like that.

Reply:
I was thinking about this a little bit more. How about supporting those types on `EvaluateExpressionAsync`?
We could have something like:

 * `EvaluateExpressionAsync`
 * `EvaluateExpressionAsync<int>`
 * `EvaluateExpressionAsync<JSHandle>`
 * `EvaluateExpressionAsync<ElementHandle>`

Reply:
It feels odd having a "Generic" method, that we know it would only work with 2 types.

Reply:
Or we could just cast. In js you don't need since it's dynamic...

Reply:
ok... I'm not buying completely. but let's see how it feels.

Reply:
Too much talk for nothing, well we learned. I need to confirm but  `Page.EvaluateExpressionHandleAsync("button")` will return an ElementHandle from Chromium.

But now I'm thinking about this method. I think we can remove the override on `ElementHandle` and implement this:
```
public virtual ElementHandle AsElement() => this as ElementHandle;
```

The purpose is the same and it's a shorter solution. They weren't able to do this in Puppeteer but we can do it here.